src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyStore.java

```diff
@@ -454,7 +455,18 @@ public synchronized void engineSetKeyEntry(String alias, Key key,
         } catch (NullPointerException | IllegalArgumentException e) {
             throw new KeyStoreException(e);
         }
-        engineSetEntry(alias, entry, new KeyStore.PasswordProtection(password));
+
+        KeyStore.PasswordProtection passwordProtection =
+                new KeyStore.PasswordProtection(password);
+        try {
+            engineSetEntry(alias, entry, passwordProtection);
+        } finally {
+            try {
+                passwordProtection.destroy();
+            } catch (DestroyFailedException dfe) {
+                // ignore
+            }
+        }
```

Comment:
This particular change looks fine. 
However, I noticed that there is another local PasswordProtection object inside the KeyStoreLoadParameter class in Secmod.java. Is that also subject to the same issue?